Source/Charts/Renderers/XAxisRenderer.swift

```diff
@@ -130,33 +116,17 @@ open class XAxisRenderer: NSObject, AxisRenderer
 
             let last = interval == 0.0 ? 0.0 : (floor(yMax / interval) * interval).nextUp
 
-            if interval != 0.0 && last != first
+            if interval != 0.0, last != first
             {
-                for _ in stride(from: first, through: last, by: interval)
-                {
-                    n += 1
-                }
+                stride(from: first, through: last, by: interval).forEach { _ in n += 1 }
             }
 
             // Ensure stops contains at least n elements.
             axis.entries.removeAll(keepingCapacity: true)
             axis.entries.reserveCapacity(labelCount)
 
-            var f = first
-            var i = 0
-            while i < n
-            {
-                if f == 0.0
-                {
-                    // Fix for IEEE negative zero case (Where value == -0.0, and 0.0 == -0.0)
-                    f = 0.0
-                }
-
-                axis.entries.append(Double(f))
-
-                f += interval
-                i += 1
-            }
+            let values = stride(from: first, to: Double(n) * interval + first, by: interval)
```

Comment:
feel kind of increased the complex to read with a glimpse

Reply:
I updated it. It's really just `y = ax + b`